src/riak_kv_pb_timeseries.erl

```diff
@@ -316,21 +321,28 @@ put_data(Data, Table, Mod) when is_binary(Table) ->
                   case riak_kv_w1c_worker:validate_options(
                          NVal, Preflist, [], BucketProps) of
                       {ok, W, PW} ->
-                          {Ids, Errs} =
-                              lists:foldl(
-                                fun(Record, {PartReqIds, PartErrors}) ->
-                                        {RObj, LK} =
-                                            build_object(Bucket, Mod, DDL,
-                                                         Record, DocIdx),
-
-                                        {ok, ReqId} =
-                                            riak_kv_w1c_worker:async_put(
-                                              RObj, W, PW, Bucket, NVal, LK,
-                                              EncodeFn, Preflist),
-                                        {[ReqId | PartReqIds], PartErrors}
-                                end,
-                                {[], 0}, Records),
-                          {GlobalReqIds ++ Ids, GlobalErrorsCnt + Errs};
+                          DataForVnode = pick_batch_option(SendFullBatches,
```

Comment:
Regrettably, I can't find a good way to fix this. There is a massive number of bindings in this function which have to be curried around, so neither the outer anonymous function for the `foldl` nor the inner anonymous functions for the `invoke_async_put` invocation can be extracted into named functions without creating a different kind of monster.

Ideas greatly encouraged, because I definitely don't like the way this turned out, speaking as a fan of one-line functions.